### ISSUE TITLE
Implement unlikelihood training

### ISSUE THREAD

**Issue by gcooper-isi** [gcooper-isi/Megatron-LM#13]
Unlikelihood training is described in the paper "Neural text (de-)generation with unlikelihood training", available at https://arxiv.org/pdf/1908.04319.pdf. The code is available at https://github.com/facebookresearch/unlikelihood_training/. However, they used the original GPT-2 rather than Megatron, so their code will need to be adapted for Megatron.

The basic problem that they are trying to solve is that GPT-2 models tend to generate repetitive text, particularly when using greedy decoding or beam search. The idea is basically to add a penalty for generating repetitive text into the loss function.
